### PR TITLE
Plains2Peaks mapping

### PR DESCRIPTION
**Pull request by moltude**
Add P2P mapping and tests

### CODE REVIEW

**Comment by moltude** [comment]
@codacy-badger you are wrong.

File: src/main/scala/dpla/ingestion3/mappers/providers/P2PMapping.scala [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+import org.json4s.JsonAST.JString
+import org.json4s.JsonDSL._
+import org.json4s.jackson.JsonMethods._
+
+class P2PMapping() extends Mapping[JValue] with IdMinter[JValue] with JsonExtractor {
+
+  // ID minting functions
+  override def useProviderName: Boolean = true
+
+  // Hard coded to prevent accidental changes to base ID
```
Comment (AudreyAltman) [inline src/main/scala/dpla/ingestion3/mappers/providers/P2PMapping.scala]: General kudos @moltude b/c you leave great comments in your code.

File: src/main/scala/dpla/ingestion3/mappers/providers/P2PMapping.scala [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+  override def preview(data: Document[JValue]): ZeroToOne[EdmWebResource] =
+    extractString(data.get \ "@graph" \ "edm:preview" \ "@id").map(new URI(_)).map(uriOnlyWebResource)
+
+  override def provider(data: Document[JValue]): ExactlyOne[EdmAgent] = EdmAgent(
+    name = Some("Plains to Peaks Collective"),
+    uri = Some(new URI("http://dp.la/api/contributor/p2p"))
+  )
+
+  override def sidecar(data: Document[JValue]): JValue =
+    ("prehashId", buildProviderBaseId()(data)) ~ ("dplaId", mintDplaId(data))
```
Comment (AudreyAltman) [inline src/main/scala/dpla/ingestion3/mappers/providers/P2PMapping.scala]: Can you remind me what the `~` means?

File: src/test/scala/dpla/ingestion3/mappers/providers/P2PMappingTest.scala [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  val jsonString: String = new FlatFileIO().readFileAsString("/p2p.json")
+  val json: Document[JValue] = Document(parse(jsonString))
+  val extractor = new P2PMapping
+
+  it should "extract provider id" in {
+    val expected = "https://plains2peaks.org/0a00aa9c-0b9e-11e8-a081-005056c00008"
+//     val p = (json.get \ "@graph").filterField {
+//       case JField("@type", JString("ore:Aggregation")) => true
+//       case _ => false
+//     }
```
Comment (AudreyAltman) [inline src/test/scala/dpla/ingestion3/mappers/providers/P2PMappingTest.scala]: Did you want to delete those commented-out lines?

**Pull request merged**